pkg/cmd/pkg/package_describe.go

```diff
@@ -12,82 +12,89 @@ import (
 	"github.com/dcos/dcos-core-cli/pkg/pluginutil"
 )
 
-func newCmdPackageDescribe(ctx api.Context) *cobra.Command {
-	var allVersions, appOnly, cliOnly, config, render bool
-	var appID, optionsPath, version string
+type describeOptions struct {
+	appID       string
+	optionsPath string
+	version     string
+
+	allVersions bool
+	appOnly     bool
+	cliOnly     bool
+	config      bool
+	render      bool
+}
 
+func newCmdPackageDescribe(ctx api.Context) *cobra.Command {
+	var opts describeOptions
 	cmd := &cobra.Command{
 		Use:   "describe <package-name>",
 		Short: "Get specific details for packages",
 		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			packageName := args[0]
-
-			c, err := cosmos.NewClient(ctx, pluginutil.HTTPClient(""))
-			if err != nil {
-				return err
-			}
-			enc := json.NewEncoder(ctx.Out())
-			enc.SetIndent("", "  ")
-
-			if allVersions {
-				versions, err := c.PackageListVersions(packageName)
-				if err != nil {
-					return err
-				}
-				return enc.Encode(versions)
+			return pkgDescribe(ctx, args[0], opts)
+		},
+	}
+	cmd.Flags().BoolVar(&opts.appOnly, "app", false, "Application only")
+	cmd.Flags().StringVar(&opts.appID, "app-id", "", "The application ID")
+	cmd.Flags().BoolVar(&opts.cliOnly, "cli", false, "Command line only")
+	cmd.Flags().BoolVar(&opts.config, "opts.config", false, "Print the opts.configurable properties of the `marathon.json` file")
+	cmd.Flags().StringVar(&opts.optionsPath, "describeOptions", "", "Path to a JSON file that contains customized package installation describeOptions")
+	cmd.Flags().StringVar(&opts.version, "package-opts.version", "", "The package opts.version")
+	cmd.Flags().BoolVar(&opts.allVersions, "package-versions", false, " Displays all versions for this package")
+	cmd.Flags().BoolVar(&opts.render, "opts.render", false,
+		"Collate the `marathon.json` package template with values from the `opts.config.json` and `--describeOptions`")
+	return cmd
+}
 
-			}
+func pkgDescribe(ctx api.Context, packageName string, opts describeOptions) error {
+	c, err := cosmos.NewClient(ctx, pluginutil.HTTPClient(""))
```

Comment:
Injecting this dependency would make it easier (or even possible in the first place) to unit test this func. We should then test the `cosmos.Client` independently from this command.